lua/gitlab/actions/draft_notes/init.lua

```diff
@@ -92,67 +98,18 @@ M.send_edits = function(note_id)
     local body = { note = text, position = the_note.position }
     job.run_job(string.format("/mr/draft_notes/%d", note_id), "PATCH", body, function(data)
       u.notify(data.message, vim.log.levels.INFO)
-      local has_position = false
-      local new_draft_notes = all_notes:map(function(note)
-        if note.id == note_id then
-          has_position = M.has_position(note)
-          note.note = text
-        end
-        return note
-      end)
-      state.DRAFT_NOTES = new_draft_notes
-      local discussions = require("gitlab.actions.discussions")
-      if has_position then
-        discussions.rebuild_discussion_tree()
-      else
-        discussions.rebuild_unlinked_discussion_tree()
-      end
-      winbar.update_winbar()
+      M.rebuild_view(unlinked)
     end)
   end
 end
 
--- This function will actually send the deletion to Gitlab when you make a selection, and re-render the tree
-M.send_deletion = function(tree)
-  local current_node = tree:get_node()
-  local note_node = common.get_note_node(tree, current_node)
-  local root_node = common.get_root_node(tree, current_node)
-
-  if note_node == nil or root_node == nil then
-    u.notify("Could not get note or root node", vim.log.levels.ERROR)
-    return
-  end
-
-  ---@type integer
-  local note_id = note_node.is_root and root_node.id or note_node.id
-
+---This function will actually send the deletion to Gitlab when you make a selection, and re-render the tree
+---@param note_id integer
+---@param unlinked boolean
+M.confirm_delete_draft_note = function(note_id, unlinked)
   job.run_job(string.format("/mr/draft_notes/%d", note_id), "DELETE", nil, function(data)
     u.notify(data.message, vim.log.levels.INFO)
-
-    local has_position = false
-    local new_draft_notes = List.new(state.DRAFT_NOTES):filter(function(note)
-      if note.id ~= note_id then
-        return true
-      else
-        has_position = M.has_position(note)
-        return false
-      end
-    end)
-
-    state.DRAFT_NOTES = new_draft_notes
-    local discussions = require("gitlab.actions.discussions")
-    if has_position then
-      discussions.rebuild_discussion_tree()
-    else
-      discussions.rebuild_unlinked_discussion_tree()
-    end
-
-    if state.settings.discussion_signs.enabled and state.DISCUSSION_DATA then
-      diagnostics.refresh_diagnostics()
-    end
-
-    winbar.update_winbar()
-    common.add_empty_titles()
```

Comment:
Notice all of this code is deleted 🥳 

It's now no longer necessary to parse the returned state, we just dumbly re-fetch data from our GET endpoints and rebuild the tree from scratch. This accounts for a lot of deleted code in this MR.